### PR TITLE
Add support for spaces (and other specials) in keys

### PR DESCRIPTION
**Pull request by apexskier**
Currently, page url generation (slugification) is just toLowerCase. In https://github.com/yannbertrand/macos-defaults/pull/146 I add some defaults that contain keys with spaces, which currently breaks.

This pulls in `slugify` to add support, and shell escapes keys in examples.

### CODE REVIEW

**Comment by apexskier** [comment]
I'm not sure what happened to the schema test - I haven't changed either the schema file or the defaults yaml files here.

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
     "yaml.schemas": {
         "./defaults.schema.json": ["/defaults*.yml"]
-    }
+    },
+    "[yaml]": {
+        "editor.rulers": [
+            130
+        ]
+    },
```
Comment (yannbertrand) [inline .vscode/settings.json:9]: This probably is the reason it breaks
```suggestion
    }
```

Comment (apexskier) [reply .vscode/settings.json:9]: you're right. I didn't realize the action uses this file directly. It [parses it as normal JSON](https://github.com/nwisbeta/validate-yaml-schema/blob/d6d0e9a1ff5fceb6f59b9751c79f899b324730fc/src/main.ts#L25) but this is technically jsonc

**Comment by yannbertrand** [comment]
LGTM

**Comment by yannbertrand** [comment]
Thanks!

**Pull request merged**